tests/command.deploy.test.js

```diff
@@ -398,6 +398,63 @@ if (process.env.NETLIFY_TEST_DISABLE_LIVE !== 'true') {
     })
   })
 
+  test.serial('should deploy functions from internal functions directory', async (t) => {
+    /* eslint-disable require-await */
+    await withSiteBuilder('site-with-internal-functions', async (builder) => {
+      await builder
+        .withNetlifyToml({
+          config: {
+            functions: { directory: 'functions' },
+          },
+        })
+        .withFunction({
+          path: 'func-1.js',
+          handler: async () => ({
+            statusCode: 200,
+            body: 'User 1',
+          }),
+        })
+        .withFunction({
+          path: 'func-2.js',
+          handler: async () => ({
+            statusCode: 200,
+            body: 'User 2',
+          }),
+        })
+        .withFunction({
+          path: 'func-2.js',
+          pathPrefix: '.netlify/functions-internal',
+          handler: async () => ({
+            statusCode: 200,
+            body: 'Internal 2',
+          }),
+        })
+        .withFunction({
+          path: 'func-3.js',
+          pathPrefix: '.netlify/functions-internal',
+          handler: async () => ({
+            statusCode: 200,
+            body: 'Internal 3',
+          }),
+        })
+        .buildAsync()
+
+      const { deploy_url: deployUrl } = await callCli(
+        ['deploy', '--build', '--json'],
+        {
+          cwd: builder.directory,
+          env: { NETLIFY_SITE_ID: t.context.siteId },
+        },
+        true,
+      )
+
+      t.is(await got(`${deployUrl}/.netlify/functions/func-1`).text(), 'User 1')
```

Comment:
[sand] Do we want to test that user provider functions override internal ones in the deploy?
Is that the expected behavior?

Reply:
We are testing that. We upload two functions named `test-2` (https://github.com/netlify/cli/pull/2954/files/3c72fb4d0f63465fde7e7222e613efdda5939466#diff-f4cfe11a0a7c59e35d96895571eecb4c3386c79f7258d6a6b9222efa6010643aR418 and https://github.com/netlify/cli/pull/2954/files/3c72fb4d0f63465fde7e7222e613efdda5939466#diff-f4cfe11a0a7c59e35d96895571eecb4c3386c79f7258d6a6b9222efa6010643aR425) and asserting that the user-defined one takes precedence.

Reply:
Oh, sorry I missed - please ignore my comment.